File: backend/src/main/java/br/edu/ifs/playifs/competition/dto/CompetitionInputDTO.java
package br.edu.ifs.playifs.competition.dto;

import br.edu.ifs.playifs.data.course.model.enums.CourseLevel;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;

@Data
@Schema(description = "DTO de entrada para criar ou atualizar uma Competição.")
public class CompetitionInputDTO {

    @Schema(description = "Nome da competição.", example = "Jogos de Verão 2025", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "O campo nome é obrigatório.")
    private String name;

    @Schema(description = "Nível dos cursos que podem participar.", example = "SUPERIOR", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotNull(message = "O campo nível é obrigatório.")
    private CourseLevel level;

    // ✅ CAMPO ADICIONADO
    @Schema(description = "Lista de IDs dos desportos que farão parte desta competição.", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotEmpty(message = "A competição deve ter pelo menos um desporto associado.")
    private List<Long> sportIds;
}

File: frontend/lib/data/models/competition/competition_input_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'competition_input_dto.freezed.dart';
part 'competition_input_dto.g.dart';

/// DTO para os dados de entrada para criar/atualizar uma Competição.
@freezed
abstract class CompetitionInputDTO with _$CompetitionInputDTO {
  /// Construtor factory para o DTO.
  const factory CompetitionInputDTO({
    required String name,
    required String level,
    // ADICIONADO: Lista de IDs de desportos (Long no Java -> int no Dart).
    required List<int> sportIds,
  }) = _CompetitionInputDTO;

  /// Construtor factory para serialização para JSON.
  factory CompetitionInputDTO.fromJson(Map<String, dynamic> json) =>
      _$CompetitionInputDTOFromJson(json);
}